andela/suyabay
# #116190399  Create users endpoint controllers

#### What does this PR do?

This PR create users endpoint controllers
#### Description of Task to be completed?
- Implement Fractal transformers for data output on users endpoints
- Add getAllUsers method to UserController
- Add getSingleUser method to UserController
- Add postUser method to UserController
- Add editUser method to UserController
#### How should this be manually tested?
- To post a user, if you are using POSTMAN type `https://suyabay-staging.herokuapp.com/api/v1/users`  in the URL box and change the verb to POST then enter the fields to be posted e.g

```
     email      suyabay@evangelist.com

     username   suyabay

     password   xxxxxxx  
```

 then you should see this response  `{  "message": "User created successfully" }`
- To update a single user information , type  `https://suyabay-staging.herokuapp.com/api/v1/users/9`  in the URL box and change the verb to PUT then press submit you should see a JSON output like below.

`{  "message": "User updated successfully" }`
- To get all users, type  `https://suyabay-staging.herokuapp.com/api/v1/users`  in the URL box and change the verb to GET then press submit you should see a JSON output like below.

```
   {
  "data": [
    {
      "id": 1,
      "username": "Sade adu",
      "email": "olotintemitop12e@gmail.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-23 18:46:42.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 2,
      "username": "Shania Hoeger",
      "email": "Rossie04@yahoo.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 3,
      "username": "Breanna Osinski II",
      "email": "Margie.West@yahoo.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 4,
      "username": "Terrence Wolff",
      "email": "Raymond13@Bernier.net",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 5,
      "username": "Edyth Stokes",
      "email": "Bins.Dell@gmail.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 6,
      "username": "Roy",
      "email": "temitope.olotin@andela.com",
      "date_created": {
        "date": "2016-03-10 08:41:34.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-23 19:17:28.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": "http://res.cloudinary.com/suyabay/image/upload/v1458735567/themzdqhwl8dpfxoutqx.jpg"
    },
    {
      "id": 9,
      "username": "laztopaz",
      "email": "olotintemitope@gmail.com",
      "date_created": {
        "date": "2016-03-23 18:15:39.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-30 10:03:18.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 10,
      "username": "codekenn",
      "email": "kenny.ogunde@andela.com",
      "date_created": {
        "date": "2016-03-30 09:09:01.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-30 09:09:01.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    }
  ]
}

```
- To get a single user , type  `https://suyabay-staging.herokuapp.com/api/v1/users/9`  in the URL box and change the verb to GET then press submit you should see a JSON output like below.

```

{
  "data": [
    {
      "id": 9,
      "username": "laztopaz",
      "email": "olotintemitope@gmail.com",
      "date_created": {
        "date": "2016-03-23 18:15:39.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-30 10:03:18.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    }
  ]
}

```
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
#116190399
#### Screenshots (if appropriate)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
